zerver/lib/queue.py

```diff
@@ -188,31 +189,32 @@ def _connect(self, on_open_cb: Optional[Callable[[], None]]=None) -> None:
         self.connection = ExceptionFreeTornadoConnection(
             self._get_parameters(),
             on_open_callback = self._on_open,
+            on_open_error_callback = self._on_connection_open_error,
             on_close_callback = self._on_connection_closed,
         )
 
     def _reconnect(self) -> None:
         self.connection = None
         self.channel = None
         self.queues = set()
+        self.log.warning("TornadoQueueClient attempting to reconnect to RabbitMQ")
         self._connect()
 
-    def _on_connection_closed(self, connection: pika.connection.Connection,
-                              reply_code: int, reply_text: str) -> None:
-        self.log.warning("TornadoQueueClient lost connection to RabbitMQ, reconnecting...")
-        from tornado import ioloop
-
-        # Try to reconnect in two seconds
-        retry_seconds = 2
+    CONNECTION_RETRY_SECS = 5
```

Comment:
I don't see much value in reducing this number.  A Zulip server is non-operational for important functionality in between retries, and there's nothing we use rabbitmq for that's similar in importance to Tornado (and there's just 1 Tornado process), so this seems like a case where we should retry aggressively in the hope of avoiding longer-than-necessary downtime.

Reply:
Sure, fair enough.